src/ophys_etl/modules/segmentation/graph_utils/edge_attributes.py

```diff
@@ -0,0 +1,381 @@
+
+    """
+    new_graph = nx.Graph()
+    # copies over node attributes
+    new_graph.add_nodes_from(graph.nodes(data=True))
+
+    # load the section of data that encompasses this graph
+    rows, cols = np.array(graph.nodes).T
+    with h5py.File(video_path, "r") as f:
+        data = f["data"][:,
+                         rows.min(): (rows.max() + 1),
+                         cols.min(): (cols.max() + 1)]
+
+    offset = np.array([rows.min(), cols.min()])
+    for node1 in graph:
+        neighbors = set(list(graph.neighbors(node1)))
+        new_neighbors = set(list(new_graph.neighbors(node1)))
+        neighbors = list(neighbors - new_neighbors)
+        if len(neighbors) == 0:
+            continue
+        nrow, ncol = np.array(node1) - offset
+        irows, icols = (np.array(neighbors) - offset).T
+        weights = 1.0 - cdist([data[:, nrow, ncol]],
+                              [data[:, r, c] for r, c in zip(irows, icols)],
+                              metric="correlation")[0]
+        for node2, weight in zip(neighbors, weights):
+            attr = graph.get_edge_data(node1, node2)
+            attr.update({attribute_name: weight})
+            new_graph.add_edge(node1, node2, **attr)
+
+    return new_graph
+
+
+def add_filtered_pearson_edge_attributes(
+        filter_fraction: float,
+        graph: nx.Graph,
+        video_path: Path,
+        attribute_name: str = "filtered_Pearson") -> nx.Graph:
+    """adds an attribute to each edge which is the Pearson correlation
+    coefficient between the traces of the two pixels (nodes) associated
+    with that edge. Correlation coefficient is only calculated for the
+    union of the brightest `filter_fraction` of timesteps at the two
+    pixels
+
+    Parameters
+    ----------
+    filter_fraction: float
+        The fraction of timesteps to keep for each pixel when calculating
+        the correlation coefficient
+    graph: nx.Graph
+        a graph with nodes like (row, col) and edges connecting them
+    video_path: Path
+        path to an hdf5 video file, assumed to have a dataset "data"
+        nframes x nrow x ncol
+    attribute_name: str
+        name set on each edge for this calculated value
+
+    Returns
+    -------
+    new_graph: nx.Graph
+        an undirected networkx graph, with attribute added to edges
+
+    """
+    new_graph = nx.Graph()
+    # copies over node attributes
+    new_graph.add_nodes_from(graph.nodes(data=True))
+
+    # load the section of data that encompasses this graph
+    rows, cols = np.array(graph.nodes).T
+    with h5py.File(video_path, "r") as f:
+        data = f["data"][:,
+                         rows.min(): (rows.max() + 1),
+                         cols.min(): (cols.max() + 1)]
+
+    offset = np.array([rows.min(), cols.min()])
+
+    # create a lookup table of the brightest filter_fraction
+    # timesteps for each pixel
+    discard = 1.0-filter_fraction
```

Comment:
I'm a bit surprised that this line doesn't cause an error with the linter: https://www.flake8rules.com/rules/E225.html

Reply:
that is surprising, i'll fix

Reply:
ah ha:
https://newbedev.com/does-pep-8-require-whitespace-around-operators-in-function-arguments

Reply:
fixed